assembly/src/assembler/id.rs

```diff
@@ -0,0 +1,49 @@
+use crate::ast::ProcedureIndex;
+
+/// Uniquely identifies a procedure in a set of [crate::ast::Module]
+///
+/// A [GlobalProcedureIndex] is assigned to a procedure when it is added to a [super::ModuleGraph].
+/// The index uniquely identifies that procedure in the graph, and provides a unique, copyable,
+/// machine-word sized handle that can be trivially stored, passed around, and later used to
+/// perform constant-complexity operations against that procedure.
+///
+/// <div class="warning">As a result of this being just an index into a specific instance of a
+/// [super::ModuleGraph], it does not provide any guarantees about uniqueness or stability when
+/// the same module is stored in multiple graphs - each graph may assign it a unique index. You
+/// must ensure that you do not store these indices and attempt to use them with just any module
+/// graph - it is only valid with the one it was assigned from.
```

Comment:
I'm not sure if this is needed. Are the docs XHTML or HTML?
```suggestion
/// <div class="warning">As a result of this being just an index into a specific instance of a
/// [super::ModuleGraph], it does not provide any guarantees about uniqueness or stability when
/// the same module is stored in multiple graphs - each graph may assign it a unique index. You
/// must ensure that you do not store these indices and attempt to use them with just any module
/// graph - it is only valid with the one it was assigned from.</div>
```

Reply:
I'm not actually sure, but I missed the closing tag here in any case, and certainly want it even if it isn't required. Good eye!